src/plugins/intel_cpu/tests/functional/subgraph_tests/src/ngram.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+#include <debug.h>
+#include <shared_test_classes/base/ov_subgraph.hpp>
+#include <ngraph_functions/builders.hpp>
+#include "common_test_utils/common_utils.hpp"
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include "test_utils/cpu_test_utils.hpp"
+#include <openvino/opsets/opset1.hpp>
+
+using namespace CPUTestUtils;
+using namespace ov::test;
+using namespace ngraph::helpers;
+
+namespace CPUSubgraphTestsDefinitions {
+
+typedef std::tuple<
+    std::vector<InputShape>,
+    size_t
+> NgramTestParams;
+
+static std::shared_ptr<ov::Node> getStridedSlice(const std::shared_ptr<ov::Node> data,
+                                                 const std::shared_ptr<ov::Node> begin,
+                                                 const std::shared_ptr<ov::Node> end,
+                                                 const std::vector<int64_t>& shrink_axis_mask = {}) {
+    std::vector<int64_t> default_mask(begin->get_shape()[0], 0);
+    return std::make_shared<ov::opset1::StridedSlice>(data, begin, end, default_mask, default_mask,
+                                                      std::vector<int64_t>{}, shrink_axis_mask);
+}
+
+static std::shared_ptr<ov::Node> getReshape(const std::shared_ptr<ov::Node> data,
+                                            const std::vector<int64_t>& requested_shape,
+                                            const ov::element::Type& prc) {
+    auto requested_shape_node = ov::opset1::Constant::create(prc, {requested_shape.size()}, requested_shape);
+    return std::make_shared<ov::opset1::Reshape>(data, requested_shape_node, true);
+}
+
+static std::shared_ptr<ov::Model> initNgram(std::vector<ov::PartialShape>& input_shapes, const size_t k) {
+    const auto data_et = ov::element::f32;
+    const auto idces_et = ov::element::i32;
```

Comment:
Could you parameterize precisions to support others in future?

Reply:
Good idea, thanks for the suggestion. Done